pkg/installmanager/installmanager.go

```diff
@@ -563,7 +563,11 @@ func cleanupFailedProvision(dynClient client.Client, cd *hivev1.ClusterDeploymen
 		uninstaller.GraphAuthorizer = session.GraphAuthorizer
 		uninstaller.Authorizer = session.Authorizer
 
-		return uninstaller.Run()
+		if err := uninstaller.Run(); err != nil {
+			return err
+		}
+
+		return cleanupDNSZone(dynClient, cd, logger)
```

Comment:
Move outside switch.

Reply:
Yes. I did that in the GCP PR, so this PR should just inherit that change in flow.